WordPress/gutenberg
# Navigation: Add integration tests for NavigationMenuSelector component

## What?


Adds a suite of integration/component level tests for the `<NavigationMenuSelector>` component in the Nav block.

Also fixes a few bugs discovered along the way.

## Why?


This component has suffered from a number of regressions. Moreover it has a number of states and potential edge cases that are too difficult for humans to manually test. This makes the component brittle and difficult to change.

Adding these tests provides an intial measure of confidence in the component's behaviour. After this it will be necessary to refactor the component to remove some of the unusual patterns that are in evidence (e.g. setState on render). This should help to simplify it but should not be done without the backing of tests.

## How?


Adds integration tests that assert on all the states of the component. Various hooks are mocked to control the state of the component. That's ok. We sacrifice some certainty/accuracy for speed and ease of use. It's gets us 80% of the coverage with 20% of the effort relative to e2e tests.



## Testing Instructions





It's critical we try and verify the original behaviour from `trunk` is maintained:

- Remove all Nav Menus and Classic Menus
- New Post -> Nav block
- Check dropdown has only "Create new menu" option.
- Check clicking on that works as expected.
- Create some more menus. Check you can switch between them and the menu updaes correctly.
- Switch to Classic Theme and create some classic menus.
- Switch back to block theme.
- Check classic menus are shown and you can successfully import them and switch between them.

With that checked you can run the test suite:

```
npm run test:unit packages/block-library/src/navigation/edit/test/navigation-menu-selector.js
```

### Testing Instructions for Keyboard


## Screenshots or screencast
